File: programming/lab2/src/main/java/moves/status/Charm.java
package moves.status;

import ru.ifmo.se.pokemon.*;

/**
 * Charm lowers the target's Attack by two stages.
 * <a href="https://pokemondb.net/move/charm">Pokemon DB description</a>
 */
public class Charm extends StatusMove {
    public Charm() {
        super(Type.FAIRY, 0, 100);
    }

    @Override
    protected boolean checkAccuracy(Pokemon pokemon, Pokemon pokemon1) {
        return true;
    }

    @Override
    protected String describe() {
        return "использует Charm";
    }

    @Override
    protected void applyOppEffects(Pokemon pokemon) {
        pokemon.addEffect(new Effect().stat(Stat.ATTACK, -2).turns(-1));
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/status/Charm.kt
package org.itmo.moves.status

import ru.ifmo.se.pokemon.*

class Charm : StatusMove(Type.FAIRY, .0, 100.0) {
    override fun applyOppEffects(p0: Pokemon) {
        p0.addEffect(Effect().turns(-1).stat(Stat.ATTACK, -2))
    }

    override fun checkAccuracy(p0: Pokemon?, p1: Pokemon?): Boolean {
        return true
    }

    override fun describe(): String {
        return "использует Charm"
    }
}